File: restaurant-frontend/src/Component/Services/MealService.js
import axios from "axios";

const MEAL_BASE_REST_API_URL= 'http://localhost:8080/api/meal';

class MealService{
    getAllMeals(){
        return axios.get(MEAL_BASE_REST_API_URL)
    }
    createMeal(image,mealName,category,price,description){
        let formData =new FormData();
        formData.append("image",image);
        formData.append("mealName",mealName);
        formData.append("price",price);
        formData.append("category",category);
        formData.append("desc",description);
        return axios.post(MEAL_BASE_REST_API_URL ,formData);
    
    }

    getMealById(meal_id){
        return axios.get(MEAL_BASE_REST_API_URL + '/' + meal_id);
    }


    updateMealImage(meal_id,image,mealName,category,price,description){
        let formData =new FormData();
        formData.append("image",image);
        formData.append("mealName",mealName);
        formData.append("price",price);
        formData.append("category",category);
        formData.append("desc",description);
        console.log(image);
     
        return axios.put(MEAL_BASE_REST_API_URL + '/' + meal_id,formData);
    }

    deleteMeal(meal_id){
        return axios.delete(MEAL_BASE_REST_API_URL + '/' + meal_id);
    }
}
export default new MealService();

File: restaurant-backend/src/main/java/com/app/restaurant/service/MealService.java
package com.app.restaurant.service;

import com.app.restaurant.dto.MealIngredientDTO;
import com.app.restaurant.exception.ResourceNotFoundException;
import com.app.restaurant.model.Ingredient;
import com.app.restaurant.model.Meal;
import com.app.restaurant.model.MealIngredient;
import com.app.restaurant.repository.IngredientRepository;
import com.app.restaurant.repository.MealIngredientrepository;
import com.app.restaurant.repository.MealRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.util.*;

@Service
public class MealService {

    @Autowired
    private MealRepository mealRepo;

    @Autowired
    MealIngredientrepository mealingredientrepo;

    @Autowired
    IngredientRepository ingredientRepo;
    public List<Meal> getAllMeals(){

        return mealRepo.findAll();
    }

    //build create meal REST API

    public Meal createMeal(MultipartFile image, String mealName, double price, String category, String description) throws IOException {
        Meal newMeal = new Meal();
        newMeal.setImage(Base64.getEncoder().encodeToString((image.getBytes())));
        newMeal.setMealName(mealName);
        newMeal.setCategory(category);
        newMeal.setPrice(price);
        newMeal.setDescription(description);
        mealRepo.save(newMeal);


        return newMeal;
    }

    //build get meal by id REST API

    public Meal getMealById(int id){
        Meal meal =mealRepo.findById(id).orElseThrow(() -> new ResourceNotFoundException("Meal not exist with id " +id));

        return meal;
    }
    public Meal updateMeal(int id, MultipartFile image,
                           String mealName,
                           double price,
                           String category,
                           String description) throws IOException {

        Meal updateMeal =mealRepo.findById(id).orElseThrow(() -> new ResourceNotFoundException("Meal not exist with id " +id));
//        String imageFileName = StringUtils.cleanPath(image.getOriginalFilename());
//        System.out.println("imagepath"+imageFileName);


        updateMeal.setImage(Base64.getEncoder().encodeToString((image.getBytes())));


        updateMeal.setMealName(mealName);
        updateMeal.setCategory(category);
        updateMeal.setPrice(price);
        updateMeal.setDescription(description);
        mealRepo.save(updateMeal);


        return updateMeal;

    }
//    public Meal updateMeal(int id, MultipartFile image,
//                           String mealName,
//                           double price,
//                           String category,
//                           String description,
//                           List<MealIngredient> mealIngredient) throws IOException {
//
//        Meal updateMeal =mealRepo.findById(id).orElseThrow(() -> new ResourceNotFoundException("Meal not exist with id " +id));
////        String imageFileName = StringUtils.cleanPath(image.getOriginalFilename());
////        System.out.println("imagepath"+imageFileName);
//        updateMeal.setImage(Base64.getEncoder().encodeToString((image.getBytes())));
//        updateMeal.setMealName(mealName);
//        updateMeal.setCategory(category);
//        updateMeal.setPrice(price);
//        updateMeal.setDescription(description);
//        updateMeal.setMealIngredients(mealIngredient);
//
//        mealRepo.save(updateMeal);
//        return updateMeal;
//
//    }

    // build delete meal REST API

    public Map<String,Boolean> deleteMeal(int id){
        Meal meal =mealRepo.findById(id).orElseThrow(() -> new ResourceNotFoundException("Meal not exist with id " +id));

        mealRepo.delete(meal);
        Map<String,Boolean> response = new HashMap<>();
        response.put("deleted",Boolean.TRUE);
        return response;
    }



}
